integration/tests/helpers/context.py

```diff
@@ -0,0 +1,46 @@
+from . import api
+from .conversions import *
+from .response import Response
+import requests
+
+class Context:
+    def __init__(self, domain, service_map, version):
+        self.domain = domain
+        self.service_map = service_map
+        self.version = version
+
+    def mkurl(self, service, path, *, internal=False, protocol='http'):
+        port = self.service_map[service]
+        if not path or path[0] != '/':
+            path = '/' + path
+        if self.version is not None:
+            vpath = f'/v{self.version}'
+        else:
+            vpath = ''
+        return f'{protocol}://localhost:{port}{vpath}{path}'
+
+    def request(self, method, url, user=None, conn_id='0', headers=None, **kwargs):
+        if headers is None:
+            headers = {}
+        if user is not None:
+            headers['Z-User'] = obj_id(user)
+            headers['Z-Connection'] = conn_id
+        return Response(method, url, kwargs, 
+                        requests.request(method, url, headers=headers, **kwargs))
+    
+    def __getattr__(self, name):
```

Comment:
Note to reviewers: This functions makes it possible to write `ctx.<any-method>` for `<any-method` that is defined in `api`. That way all api methods don't have to be defined as methods of `Context`

Reply:
How about copying this block of text as a comment above this function for future readers? 🙃